fili-core/src/main/java/com/yahoo/bard/webservice/data/config/metric/makers/MetricMaker.java

```diff
@@ -149,23 +153,17 @@ protected TemplateDruidQuery getDependentQuery(String name) {
      * @return The merged query
      */
     protected TemplateDruidQuery getMergedQuery(List<String> names) {
-        if (names.size() <= 0) {
-            String message = "At least 1 name is needed to merge aggregations";
-            LOG.error(message);
-            throw new IllegalArgumentException(message);
-        }
-
-        // Get the initial query
-        TemplateDruidQuery query = getDependentQuery(names.get(0));
-
         // Merge in any additional queries
-        for (int i = 1; i < names.size(); i++) {
-            TemplateDruidQuery additionalQuery = getDependentQuery(names.get(i));
-            query = query.merge(additionalQuery);
-        }
-
-        // Return the merged query
-        return query;
+        return names.stream()
+                .map(metrics::get)
+                .map(LogicalMetric::getTemplateDruidQuery)
+                .reduce(TemplateDruidQuery::merge)
+                .orElseGet(() -> {
+                            String message = "At least 1 name is needed to merge aggregations";
+                            LOG.error(message);
+                            throw new IllegalArgumentException(message);
+                        }
```

Comment:
Autoformat thinks that the closure shouldn't be double indented because it's a sub block not an argument line continuation.  What do you think of what I'm about to post?